lms/djangoapps/courseware/tests/test_access.py

```diff
@@ -327,6 +384,13 @@ def test_access_on_course_with_pre_requisites(self):
         fulfill_course_milestone(pre_requisite_course.id, user)
         self.assertTrue(access._has_access_course_desc(user, 'view_courseware_with_prerequisites', course))
 
+    def test__access_on_mobile(self):
```

Comment:
Is there a test that verifies that non-staff members cannot bypass the mobile_available flag?

Consider using `ddt` to test the various combinations in these test cases.